ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -528,10 +536,97 @@ protected void initTreeTable() {
     doInit();
   }
 
+  private static Comparator<Task> beginComparatorAsc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return t1.getStart().compareTo(t2.getStart());
+    }
+  };
+
+  private static Comparator<Task> beginComparatorDesc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return -t1.getStart().compareTo(t2.getStart());
+    }
+  };
+
+  private static Comparator<Task> endComparatorAsc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return t1.getEnd().compareTo(t2.getEnd());
+    }
+  };
+
+  private Comparator<Task> endComparatorDesc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return -t1.getEnd().compareTo(t2.getEnd());
+    }
+  };
+
+  private class SortTableHeaderRenderer implements TableCellRenderer {
+    private Icon ascIcon;
+    private Icon descIcon;
+    private TableCellRenderer defaultRenderer;
+    SortTableHeaderRenderer(JTable t) {
+      defaultRenderer = t.getTableHeader().getDefaultRenderer();
+      ascIcon = UIManager.getIcon("Table.ascendingSortIcon");
+      descIcon = UIManager.getIcon("Table.descendingSortIcon");
+    }
+
+    @Override
+    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int col) {
+      Component c = defaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, col);
+      ColumnImpl column = myTableHeaderFacade.findColumnByViewIndex(col);
+      if (column.getSort() == SortOrder.ASCENDING) {
+        ((JLabel) c).setIcon(ascIcon);
```

Comment:
Please add `instanceof JLabel` check. If default component is not JLabel then we just ignore it (or maybe replace with our own JLabel?)